src/Symfony/Component/Security/Http/Firewall/ContextListener.php

```diff
@@ -292,7 +292,7 @@ protected function refreshUser(TokenInterface $token)
             return null;
         }
 
-        throw new \RuntimeException(sprintf('There is no user provider for user "%s".', $userClass));
+        throw new UnsupportedUserException(sprintf('There is no user provider for user "%s.%s', $userClass, count($this->userProviders) > 0 ? ' Are you sure supportsClass() of your user provider returns true for this classname?' : ''));
```

Comment:
Reaching this exception means you have to fix your code, a `\LogicException` would be a better a fit. 
Also we should double check that this change does not impact the current error handling given it's an `AuthenticationServiceException` (is there anything likely to catch it and perform some special treatment regarding e.g. logging or response handling?)

Reply:
Oh, I just was a lazy copy paster and didn't even notice this used a different class. I've reverted it back to `\RuntimeException`, it wasn't my intention to change it.